### PR TITLE
fixes for comma as decimal separator

### PR DESCRIPTION
**Pull request by kaggrwal**
Fixes for #640

### CODE REVIEW

File: src/Components/DonationWidget.js [hunk 1]
```diff
@@ -84,13 +84,11 @@ export default class DonationWidget extends Component {
                         <div className="input-addon-container">
                             <input
                                 id="donation-widget-amount"
-                                type="number"
```
Comment (mal-tee) [inline src/Components/DonationWidget.js:87]: We don't really want to change that.

Comment (kaggrwal) [reply src/Components/DonationWidget.js:87]: But you can't identify the comma then, It can only be taken as input when it is text.

Comment (baltpeter) [reply src/Components/DonationWidget.js:87]: I've dug into this a little: The `number` input isn't a problem. If you set the locale to German in Firefox, a regular `<input type="number">` accepts commas just fine. Somehow our code breaks this.

Comment (mal-tee) [reply src/Components/DonationWidget.js:87]: seems like you are right: https://jsfiddle.net/r9L7bj4a/7/

Comment (baltpeter) [reply src/Components/DonationWidget.js:87]: You can actually even debug this without setting your browser to German: Even in English, entering a decimal with a period doesn't work. It's likely that that's the same problem.

Comment (kaggrwal) [reply src/Components/DonationWidget.js:87]: I tried changing the locale in google chrome, but I couldn't input a comma.

Comment (baltpeter) [reply src/Components/DonationWidget.js:87]: Works fine for me in both Firefox and Chrome:

https://user-images.githubusercontent.com/4048809/135908901-348b3d73-03fa-4d65-97d5-62f87128d365.mp4

Comment (kaggrwal) [reply src/Components/DonationWidget.js:87]: Are you using any extension ?
 I used some extension to change the locale on my google chrome.

Comment (baltpeter) [reply src/Components/DonationWidget.js:87]: No, that is a native installation on a German system. The extension probably isn't the right way to change the locale.

But as I said, for the issue, you can focus on the period as the decimal separator.

Comment (davenewham) [reply src/Components/DonationWidget.js:87]: The reason the website breaks decimal input is that it is parsed as a float on input, so "10," gets cut to 10 meaning that you can never add a decimal, but if you use the stepper, the input becomes "10,01" which is valid and is then parsed correctly.

Other than removing this parsing ball together, I think it's the correct way of doing it. The || 0 needs to be omitted. However, this code chunk comes from StackOverflow so will need correct attribution

Comment (kaggrwal) [reply src/Components/DonationWidget.js:87]: So, you mean you just want to treat period "."  as a decimal operator. 
Correct me if am wrong, but the period is already treated as a decimal separator.

Comment (baltpeter) [reply src/Components/DonationWidget.js:87]: No, it's not. If you try to enter a decimal point in the donate form, it is silently discarded.

**Comment by baltpeter** [comment]
Hey @kaggrwal, are you still working on this or do you want me to close the PR and unassign you from the issue?

**Pull request closed without merging**